nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -413,17 +418,36 @@ private static void reportInvalidOverridingMethodParamTypeError(
    * @param analysis the analysis object
    * @param state the visitor state
    */
-  public static void checkTypeParameterNullnessForAssignability(
+  public void checkTypeParameterNullnessForAssignability(
       Tree tree, NullAway analysis, VisitorState state) {
     Config config = analysis.getConfig();
     if (!config.isJSpecifyMode()) {
       return;
     }
     Type lhsType = getTreeType(tree, config);
+    if (lhsType == null) {
+      return;
+    }
     Tree rhsTree;
     if (tree instanceof VariableTree) {
       VariableTree varTree = (VariableTree) tree;
       rhsTree = varTree.getInitializer();
+
+      if (rhsTree instanceof MethodInvocationTree) {
```

Comment:
I think this logic is not in the right place, as it won't work for regular assignments; it only runs for variable declarations.  I added a failing test to illustrate.